<?xml version="1.0" ?>
<launch>

    <arg name="nicla_name" default="nicla" />
    <arg name="receiver_ip"/>
    <arg name="receiver_port" default="8002" />
    <arg name="connection_type"/> <!-- tcp, udp -->
    <arg name="enable_range" default="true" />
    <arg name="enable_camera_raw" default="false" />
    <arg name="enable_camera_compressed" default="true" />
    <arg name="enable_audio" default="false" />
    <arg name="enable_audio_stamped" default="true" />
    <arg name="enable_imu" default="true" />
    
    <node name="nicla_receiver" pkg="nicla_vision_ros2" exec="nicla_receiver" output="screen" respawn="false" >
        <param name="nicla_name" value="$(var nicla_name)" />
        <param name="receiver_ip" value="$(var receiver_ip)" />
        <param name="receiver_port" value="$(var receiver_port)" />
        <param name="connection_type" value="$(var connection_type)" />

        <param name="enable_range" value="$(var enable_range)" />
        <param name="enable_camera_raw" value="$(var enable_camera_raw)" />
        <param name="enable_camera_compressed" value="$(var enable_camera_compressed)" />
        <param name="enable_audio" value="$(var enable_audio)" />
        <param name="enable_audio_stamped" value="$(var enable_audio_stamped)" />
        <param name="enable_imu" value="$(var enable_imu)" />
    </node>

</launch>
